plotter/db.py

```diff
@@ -166,20 +166,28 @@ def close_db():
     env.close()
 atexit.register(close_db)
 
+DB_CLASSES = []
 
 class DB(type):
-
+    """Metaclass for Resource objects"""
     def __init__(cls, name, bases, dct):
-        cls.filename = name
-        cls.db = bsddb3.db.DB(env)
-        if cls.RE_LEN:
-            cls.db.set_re_len(cls.RE_LEN)
-        cls.db.open(cls.filename, None, cls.DBTYPE,
-                    bsddb3.db.DB_AUTO_COMMIT |
-                    # bsddb3.db.DB_THREAD|
-                    bsddb3.db.DB_CREATE)
-        CLOSE_ON_EXIT.append(cls.db)
-
+        """Called when Resource and each subclass is defined"""
+        if "keys" in dir(cls):
+            DB_CLASSES.append(cls)
+            cls.filename = name
+            cls.db = bsddb3.db.DB(env)
+            if cls.RE_LEN:
+                cls.db.set_re_len(cls.RE_LEN)
+            cls.db.open(cls.filename, None, cls.DBTYPE,
+                        bsddb3.db.DB_AUTO_COMMIT |
+                        # bsddb3.db.DB_THREAD|
+                        bsddb3.db.DB_CREATE)
+            CLOSE_ON_EXIT.append(cls.db)
+
+def rename_all(find, replace):
+    for cls in DB_CLASSES:
+        if all([k in cls.keys for k in find.keys()]):
+            cls.rename_all(find, replace)
```

Comment:
`rename_all` is a new function that I used to rename all database entries with keys containing old Persona usernames (e.g. someone@not-gmail.com) to new Google account usernames (e.g. someone@gmail.com)